packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/Relations.tsx

```diff
@@ -1450,11 +1439,12 @@ const UnstableListItem = ({ data, index, style }: ListItemProps) => {
     name,
     relations,
     targetModel,
-    isModalOpen,
     changeCurrentRelation,
   } = data;
-  const { formatMessage } = useIntl();
   const [showModal, setShowModal] = React.useState(false);
+  const { formatMessage } = useIntl();
+  const isModalOpen = useRelationContext('RelationContext', (state) => state.isModalOpen);
+  const setIsModalOpen = useRelationContext('RelationContext', (state) => state.setIsModalOpen);
```

Comment:
Why do we have an internal state and the state being managed by the context? Do we need to have the modal state in the context?

Reply:
we need the modal state in the context in other part of the application because I need to distinguish if it is the modal is already opened or not (for example it will be useful for the back button history)
If, at the end of the implementation the context var is useless I can remove it

Reply:
Do we need it in other places now? Do we need both the local state and the context state here in this component?

Reply:
I need the local state because otherwise the modal has the rerendering issue as you know but I need to keep track in the context if the modal is open or not.